plugins/source/gcp/codegen/recipes/aiplatform.go

```diff
@@ -0,0 +1,314 @@
+package recipes
+
+import (
+	"reflect"
+	"runtime"
+	"strings"
+
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
+	pb "cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+	longrunningpb "cloud.google.com/go/longrunning/autogen/longrunningpb"
+	locationpb "google.golang.org/genproto/googleapis/cloud/location"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func getLocationsServiceName(f interface{}) string {
+	fName := strings.Split(runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name(), "apiv1.")
+	return strings.ToLower(strings.TrimSuffix(strings.TrimPrefix(fName[1], "New"), "Client"))
+}
+
+func isFieldMockable(field reflect.StructField) bool {
+	nonMockables := []any{&structpb.Value{}, &structpb.Struct{}, &pb.Model{}, &pb.PipelineJob_RuntimeConfig{}}
```

Comment:
Faker can't mock these as we need to set the fields in a specific way so they can be `MarshalJSON`ed